mentorship ios/Views/Home/SupportingViews/DetailListCell.swift

```diff
@@ -24,7 +24,12 @@ struct DetailListCell: View {
     //alert action,. To accept, delete, reject, or withdraw a request
     func alertAction() {
         guard let reqID = self.requestData.id else { return }
-        self.relationRequestActionAPI.actOnPendingRequest(action: self.actionType, reqID: reqID)
+        self.relationRequestActionAPI.actOnPendingRequest(action: self.actionType, reqID: reqID) { message in
+            print(message)
+            if NetworkManager.responseCode == 200 {
```

Comment:
I think this should not be in a view, but network module.

Reply:
network module?

Reply:
Data source provider module it does not have to be network module.

Reply:
DetailListCell is a view and it check against network error code. This is incorrect.

Reply:
Okay

Reply:
@annabauza updated.

Reply:
@annabauza have updated the code. Completion handler block shifted to view from view model. Also, the RequestActionViewModel seems like an unnecessary overhead now. Protocol created too.

Reply:
why do I still see this ` if NetworkManager.responseCode == 200 {` in View?

Reply:
It is the completion handler code. Going by your instructions, I've shifted that from the view model to the view.

Reply:
ViewModel has no logic, NetworkServis has specific network implementation, View has ServiceProtocol and ViewModel. Service return ViewModel in View and assign it to it's local ViewModel variable.

Reply:
I don't understand how this can work in the case of multiple services being used in a view. 
Example: https://anitab-org.zulipchat.com/#narrow/stream/222534-mentorship-system/topic/iOS/near/203981497

Reply:
Because you hold declaration of ServiceProtocol not NetworkService. NetworkService is a definition of ServiceProtocol, so:
```
View {
  var service:ServiceProtocol
  var viewModel: ViewModel
  fun injectService(service:ServiceProtocol){
    self. service = service
    self.service.fetch(args,(model:Model)-> { //please note Model is not network model, it's middleware class
       self.viewModel = model.mapToViewModel()
    })
  }
}


NetworkService:ServiceProtocol{...}
DatabaseService:ServiceProtocol{...}

view.injectService(NetworkService())
view.injectService(DatabaseService())

or
SyncedLocalAndNetworkService:ServiceProtocol{
  networksService:NetworkService
  databaseService:DatabaseService
}

view.injectService(SyncedLocalAndNetworkService())

```

Does it makes more sense now?

Reply:
Yes, a little more clear now. However, I'm not clear about the working of mapToViewModel() method. Till now, we've had a model object as a property inside a view model. Also, with multiple services being used together for a view (eg. profile and home both for Home screen), how will this work?

Reply:
Write this as extension class.

Reply:
https://docs.swift.org/swift-book/LanguageGuide/Extensions.html

Reply:
I'm not clear. What should be the extension class, what should it do, and what should it extend?

Reply:
```
extension Model {
  fun mapModel() : ViewModel {
    ...
  }
}
```

Reply:
maybe it's better to call it `mapToViewModel` - so we will have instances like `mapToUserModel`, is it clear now?

Reply:
Okay. Does using an extension give any specific advantage against directly adding the method to the model class?

Also, I'm not clear about the function definition of mapModel(). Will it simply assign values to the properties of the view model? And, for that, will the model hold a reference to the view model?